packages/test-runner-selenium/src/seleniumLauncher.ts

```diff
@@ -96,6 +102,25 @@ export class SeleniumLauncher implements BrowserLauncher {
 
     return this.iframeManager;
   }
+
+  private heartbeat() {
+    const driver = this.driver;
+
+    // Heartbeat function to keep alive sessions on Sauce Labs via WebDriver calls
+    this.pendingHeartbeat = setTimeout(async () => {
```

Comment:
Should this be a setInterval?

Reply:
IIRC the recursive `setTimeout` is a bit more precise but in this case it probably doesn't matter. Changed.

Reply:
Yeah recursive would be fine too, but it wasn't recursive from what I saw. It was started when a session starts, but if a session takes a few minutes to start it would only trigger one heartbeat.